Mathlib/Data/Vector/Basic.lean

```diff
@@ -511,9 +511,25 @@ def casesOn₃ {motive : ∀ {n}, Vector α n → Vector β n → Vector γ n 
     motive v₁ v₂ v₃ :=
   inductionOn₃ (C := motive) v₁ v₂ v₃ nil @fun _ x y z xs ys zs _ => cons x y z xs ys zs
 
-/-- Cast a vector to an array. -/
-def toArray : Vector α n → Array α
-  | ⟨xs, _⟩ => cast (by rfl) xs.toArray
+/-- Convert a `List.Vector` to a `Vector`. -/
+def toVector (v : List.Vector α n) : _root_.Vector α n := ⟨v.1.toArray, v.2⟩
+
+/-- Convert a `Vector` to a `List.Vector`. -/
+def ofVector {α : Type*} {n : ℕ} (v : _root_.Vector α n) : List.Vector α n := ⟨v.toList, v.2⟩
+
+@[simp] theorem ofVector_toVector (v : List.Vector α n) : ofVector (toVector v) = v := rfl
+@[simp] theorem toVector_ofVector (v : _root_.Vector α n) : toVector (ofVector v) = v := rfl
+
+/-- The equivalence between `Vector` and `List.Vector`. -/
+@[simps!]
+def equivVector : Vector α n ≃ _root_.Vector α n where
+  toFun := toVector
+  invFun := ofVector
+  left_inv := ofVector_toVector
+  right_inv := toVector_ofVector
+
+/-- Convert a vector to an Array. -/
+def toArray : Vector α n → Array α := _root_.Vector.toArray ∘ toVector
```

Comment:
I don't think we want this double conversion, it's just one more thing to write lemmas about.

Reply:
I agree I think but we did already have it, so I thought I would put it in and then remove it if you told me too.

Reply:
Done.